### PR TITLE
Revert "RF: Adapt to datalad release 0.12.0.rc3"

### PR DESCRIPTION
**Pull request by bpoldrack**
Reverts datalad/datalad-neuroimaging#63

Reverted in favor of waiting for changes to make it into non-rc datalad release.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def findsome(subdir, extensions):
     packages=[pkg for pkg in find_packages('.') if pkg.startswith('datalad')],
     # datalad command suite specs from here
     install_requires=[
-        'datalad[full]>=0.12.0.rc3',
+        'datalad[full]>=0.10.0.rc1',
```
Comment (yarikoptic) [inline setup.py]: if you decide to go this route, please adjust this along with it to be `>= 0.11` so nobody pulls in rc's with it.

Comment (bpoldrack) [reply setup.py]: Agree.

Comment (bpoldrack) [reply setup.py]:
```suggestion
        'datalad[full]>=0.11',
```

**Comment by bpoldrack** [comment]
Okay. This is not gonna work.
@yarikoptic , @kyleam : Here I tried to revert the merge adapting neuroimaging to 0.12.0rc3, since "not needed" as discussed yesterday. But: neuroimaging already depended on a rc before. This leads to pip installing 0.12.0rc3 even if I revert. And this in turn requires adaption. I don't see a way around it. Not adapting to 0.12.0rc3 leads to failure due to changes and changing dependency to 0.11 (like Yarik suggested and I initially thought was the right thing) fails just differently. So, I lean towards not reverting at all and keep the adaptions.

**Comment by yarikoptic** [comment]
I was about to outline the rationale to just keep it dependent on rc, but then had a change of hearts. Given that some other folks (.ca?) might use it and we had other useful changes (e.g. bids procedure), and because seems to concern only with tests, I would vote to try proposed by @kyleam changes, and if that is it -- revert it to depend on a released version.  And then cut a release.
Whenever 0.12 comes out it should be easy to adjust tests to use `ok_clean_git`. @bpoldrack please let me know if you are willing to take a crack at it.

Cheers!

**Comment by yarikoptic** [comment]
grr... github ate my long comment (for better or for worse).  So just a short version -- let's try to get to work again with released version. @bpoldrack - would you be interested to give it a crack following @kyleam 's notes above?  whenever 0.12 is released it should be easy to adjust the tests to use `ok_clean_git`

**Comment by yarikoptic** [comment]
grr... github ate my long comment (for better or for worse).  So just a short version -- let's try to get to work again with released version. @bpoldrack - would you be interested to give it a crack following @kyleam 's notes above?  whenever 0.12 is released it should be easy to adjust the tests to use `ok_clean_git`

**Comment by yarikoptic** [comment]
grr... github ate my long comment (for better or for worse) and still refuses to post from chromium.  So just a short version -- let's try to get to work again with released version. @bpoldrack - would you be interested to give it a crack following @kyleam 's notes above?  whenever 0.12 is released it should be easy to adjust the tests to use `ok_clean_git`

**Comment by yarikoptic** [comment]
grr... github ate my long comment (for better or for worse).  So just a short version -- let's try to get to work again with released version. @bpoldrack - would you be interested to give it a crack following @kyleam 's notes above?  whenever 0.12 is released it should be easy to adjust the tests to use `ok_clean_git`

**Comment by yarikoptic** [comment]
grr... github ate my long comment (for better or for worse).  So just a short version -- let's try to get to work again with released version. @bpoldrack - would you be interested to give it a crack following @kyleam 's notes above?  whenever 0.12 is released it should be easy to adjust the tests to use `ok_clean_git`

**Comment by bpoldrack** [comment]
Somehow comment order is messed up. ;-)
Okay, I'll give it a try @kyleam and @yarikoptic .

**Comment by kyleam** [comment]
@yarikoptic asked me to say he is trying to post but none of his comments are showing up (same actually happened to me a few months back).  There might end up being a flood of comments from all the tries :]

**Comment by bpoldrack** [comment]
@kyleam , @yarikoptic : The comments DO show up. But the order is messed up due to some TZ confusion on github's side. Some, but not all, of your comments are shown to me as being posted 6 hours in the future and they are sorted according to those misinterpreted timestamps.

**Comment by bpoldrack** [comment]
FTR: Even my own comments are partially shown as being in the future.

**Comment by kyleam** [comment]
re: [latest failing tests][0].  It looks like the cfg_bids rewrite in c5ffc22 (ENH: Idempotent BIDS config procedure, 2019-04-26) used dataset.pathobj, which of course needs 0.12.x.  Please just go with your original idea of depending on the rc rather than spending anymore time on this.

[0]: https://travis-ci.org/datalad/datalad-neuroimaging/jobs/529829775#L1668

**Comment by yarikoptic** [comment]
> re: [latest failing tests](https://travis-ci.org/datalad/datalad-neuroimaging/jobs/529829775#L1668). It looks like the cfg_bids rewrite in [c5ffc22](https://github.com/datalad/datalad-neuroimaging/commit/c5ffc22d5f7360620fac7cbc4f1729d5ff0e5f31) (ENH: Idempotent BIDS config procedure, 2019-04-26) used dataset.pathobj, which of course needs 0.12.x. Please just go with your original idea of depending on the rc rather than spending anymore time on this.

github seems to be not healthy -- comments history is non-linear etc.
If that is the only place where pathobj is used, could be easily worked around with no loss of performance or code readability.  I will push a commit...

**Comment by bpoldrack** [comment]
Thx, @yarikoptic . Looks good.

**Comment by kyleam** [comment]
> And an older dependence on RCs is coming from the expectation that `rev_create` is available from datalad itself.

Looks like the first and only `rev_create` occurrence was added in the same commit that introduced `assert_repo_status`.

> So, my conclusion is to keep PR #63 merged and thereby make the dependence explicit and fix for rc3.

AFAICS `s/rev_create/create/` and `s/assert_repo_status/ok_clean_git/` in that test file would make it compatible with 0.11.x, but I understand if you don't want to do that because `ok_clean_git` will eventually be retired in favor of the better `assert_repo_status` (though, `ok_clean_git` is already used in many other tests in the -neuroimaging codebase, so it would just be part of a larger search-and-replace).

But really either way is fine by me.

**Pull request merged**